logics/impls/psp34_traits.rs

```diff
@@ -0,0 +1,35 @@
+use ink::prelude::{
+    string::{
+        String,
+        ToString,
+    },
+};
+use openbrush::{
+    contracts::psp34::extensions::{
+        metadata::{Internal, PSP34MetadataImpl},
+    },
+    contracts::psp34::Id,
+};
+
+use crate::traits::error::Error;
+
+#[openbrush::trait_definition]
+pub trait Psp34Traits: PSP34MetadataImpl + Internal {
+
+    /// Change baseURI
+    #[ink(message)]
+    fn set_base_uri(&mut self, uri: String) -> Result<(), Error> {
+        self._set_attribute(Id::U8(0), String::from("baseURI"), uri);
```

Comment:
I guess that `Id::U8(0)` is a convention for identifying the collection when the contrat implements `PSP34Enumerable`?

Reply:
The fact is that metadata are always linked to an `id` - I guess intended used is to have metadata linked to each token instance. So yes it's a convention, or rather an implementation details, to link metadata to the whole smartcontract.

Reply:
Metadata linked to collection_id as discussed.